torch/_inductor/fx_passes/quantization.py

```diff
@@ -896,6 +985,121 @@ def __init__(
+            binary_replace_float_out_patterns = {
+                BinaryUnaryAttr(
+                    "add", 1.0, "relu", [], ""
+                ): generate_pattern_with_unary(
+                    generate_pattern_with_binary(
+                        aten.add.Tensor,
+                        get_qlinear_pt2e_pattern(x_scale_zp_are_tensors),
+                        KeywordArg("accum_after_dequant"),
+                        int8_mixed_bf16_with_inplace_add,
+                    ),
+                    aten.relu.default,
+                ),
+            }
+
+            for (
+                binary_unary_attr,
+                patterns,
+            ) in binary_replace_float_out_patterns.items():
+                if int8_mixed_bf16_with_inplace_add:
+                    _register_quantized_linear_binary_lowering(
+                        patterns,
+                        0,  # pass_number
+                        qlinear_binary_op,  # computation_op
+                        # Note that for int8-mixed-bf16 and non-inplace add, because we have
+                        # q-dq inserted at extra input of add, so the non-inplace add has bf16 and fp32 inputs,
+                        # the output dtype will be float32.
+                        # For inplace add, there is a extra to_bf16 node at add output, so the fusion pattern has bfloat16 output.
+                        torch.bfloat16,
+                        binary_unary_attr,  # binary_unary_attr
+                    )
+                else:
+                    _register_quantized_linear_binary_lowering(
+                        patterns,
+                        1,  # pass_number
+                        qlinear_binary_op,  # computation_op
+                        torch.float32,
+                        binary_unary_attr,  # binary_unary_attr
+                    )
+
+            # Priority 3: QLinear Binary pattern with fp32/bfloat16 output
+            binary_replace_float_out_patterns = {
+                BinaryUnaryAttr(
+                    "add", 1.0, "none", [], ""
+                ): generate_pattern_with_binary(
+                    aten.add.Tensor,
+                    get_qlinear_pt2e_pattern(x_scale_zp_are_tensors),
+                    KeywordArg("accum_after_dequant"),
+                    int8_mixed_bf16_with_inplace_add,
+                ),
+            }
+
+            for (
+                binary_unary_attr,
+                patterns,
+            ) in binary_replace_float_out_patterns.items():
+                _register_quantized_linear_binary_lowering(
+                    patterns,
+                    1 if int8_mixed_bf16_with_inplace_add else 2,  # pass_number
+                    qlinear_binary_op,  # computation_op
+                    # Same output dtype setting as linear-add-relu pattern
+                    torch.bfloat16
+                    if int8_mixed_bf16_with_inplace_add
+                    else torch.float32,
```

Comment:
Same as above for case
```
dq
to_bf16
linear     extra_input(bf16)
        add
```
The output should be bf16 instead of fp32.

Reply:
Thanks. I think both fp32 and bf16 are covered.